pkg/paperwork/generator.go

```
@@ -221,15 +225,56 @@ var contentTypeToImageType = map[string]string{
 	"image/png":  "PNG",
 }
 
+func MaybeRotateImage(file afero.File, g *Generator, contentType string) (imgFile afero.File, width float64, height float64, err error) {
+	// Figure out if the image should be rotated by calculating height and width of image.
+	pic, _, decodeErr := image.Decode(file)
+	if decodeErr != nil {
+		return nil, 0.0, 0.0, errors.Wrapf(decodeErr, "file %s was not decodable", file.Name())
+	}
+	rect := pic.Bounds()
+	w := float64(rect.Max.X - rect.Min.X)
+	h := float64(rect.Max.Y - rect.Min.Y)
+
+	// If the image is landscape, then turn it to portrait orientation
+	if w > h {
+		newFile, newTemplateFileErr := g.newTempFile()
+		if newTemplateFileErr != nil {
+			return nil, 0.0, 0.0, errors.Wrap(newTemplateFileErr, "Creating temp file for image rotation")
+		}
+
+		// Rotate and save new file
+		newPic := imaging.Rotate90(pic)
+		if contentType == "image/png" {
+			png.Encode(newFile, newPic)
```

sorry just noticed this. both these errors are just being swallowed here. We should be checking these for both `png.Encode` and `jpeg.Encode`